mods/mc_worldmanager/realm/realmPlayerManagement.lua

```diff
@@ -1,8 +1,16 @@
 ---@public
 ---TeleportPlayer
 ---Teleports a player to this realm.
----@param player objectRef
+---@param player
 function Realm:TeleportPlayer(player)
+    local realmCategory = self:getCategory()
+
+    local joinable, reason = realmCategory.joinable(self, player)
+
+    if (not joinable and not minetest.check_player_privs(player, { teacher = true })) then
```

Comment:
Okay, I see this now. So if I understand correctly, the teacher has to be added as either a student or an owner of the realm?

Reply:
See my above response to previous comment for context.

While this is true, it's not necessary. Players with the teacher privilege are not checked with this system under the assumption that calling functions will handle these players separately.